<launch>
  
  <arg name="simulate" default="true" />
  <arg name="visualize" default="true" />
  <arg name="interactive" default="true" />
  <arg name="teleop" default="true" />  
  <arg name="enable-pr2" default="true" />
  <arg name="enable-bandit" default="true" />

    <!-- Remappings -->
  <remap from="/joint_state_retargeter/source_joint_states" to="/joint_states" />
  <remap from="/bandit/target_joints" to="/joint_state_retargeter/retargeted_joint_states"/>
  
  <!-- Kinematic retargeter node  -->
  <include file="$(find rss_demo)/launch/joint_state_retargeter.launch" />

  <!-- Transforms between the robots -->
  <node pkg="tf" type="static_transform_publisher" name="world_to_pr2_tf" args="0 0 0 0 0 0 /world /odom_combined 100" />
  <node pkg="tf" type="static_transform_publisher" name="world_to_bandit_tf" args="2 0 0 0 0 0 /world /torso_link 100" />
  
  <!-- PR2 -->
  <!-- TODO: WHY IS THIS BROKEN? -->
  <!-- <group ns="pr2" if="$(arg enable-pr2)" > -->
  <group if="$(arg enable-pr2)" >

    <include file="$(find rss_demo)/launch/pr2.launch" >
      <arg name="simulate" value="$(arg simulate)" />
      <arg name="visualize" value="$(arg visualize)" />
      <arg name="interactive" value="$(arg interactive)" />  
      <arg name="teleop" value="$(arg teleop)" />
    </include>
  </group>
  
  <!-- Bandit -->
  <group ns="bandit" if="$(arg enable-bandit)" >
    <include file="$(find rss_demo)/launch/bandit.launch" >
      <arg name="simulate" value="$(arg simulate)" />
      <arg name="visualize" value="false" />
      <arg name="teleop" value="$(arg teleop)" />
    </include>
  </group>
  
  <group if="$(arg visualize)" >
    <node pkg="rviz" name="rviz" type="rviz" />
  </group>
	 
</launch>